WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -81,12 +81,26 @@ def edit
   end
 
   def destroy
-    @registration = Registration.find(params[:id])
-    @registration.destroy
-    mailer = RegistrationsMailer.notify_registrant_of_deleted_registration(@registration)
-    mailer.deliver_now
-    flash[:success] = "Deleted registration and emailed #{mailer.to.join(" ")}"
-    redirect_to competition_edit_registrations_path(@registration.competition)
+    if !current_user
+      return
+    end
+
+    @competition = competition_from_params
+    is_user = params.has_key?(:user_is_deleting_theirself) || false
+    if is_user
+      registrations = @competition.registrations
+      @registration = registrations.find_by_user_id(current_user.id)
+      @registration.destroy
```

Comment:
This was a bug in the old code, so not your fault, but this should actually be `@registration.destroy!`. `.destroy` just returns true or false based on if the delete succeeded, whereas `.destroy!` will throw an error of the destroy failed.